pkg/controllers/cluster/kubeconfig_controller.go

```diff
@@ -0,0 +1,257 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Greenhouse contributors
+// SPDX-License-Identifier: Apache-2.0
+
+package cluster
+
+import (
+	"context"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/tools/clientcmd"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+
+	greenhouseapis "github.com/cloudoperators/greenhouse/pkg/apis"
+	"github.com/cloudoperators/greenhouse/pkg/apis/greenhouse/v1alpha1"
+	"github.com/cloudoperators/greenhouse/pkg/clientutil"
+)
+
+type KubeconfigReconciler struct {
+	client.Client
+}
+
+//+kubebuilder:rbac:groups=greenhouse.sap,resources=clusters,verbs=get;list;watch
+//+kubebuilder:rbac:groups=greenhouse.sap,resources=organizations,verbs=get;list;watch
+//+kubebuilder:rbac:groups="",resources=secrets,verbs=get;list;watch
+// +kubebuilder:rbac:groups=greenhouse.sap,resources=cluster-kubeconfigs,verbs=get;list;watch;create;update;patch
+
+// SetupWithManager sets up the controller with the Manager
+func (r *KubeconfigReconciler) SetupWithManager(name string, mgr ctrl.Manager) error {
+	r.Client = mgr.GetClient()
+
+	return ctrl.NewControllerManagedBy(mgr).
+		Named(name).
+		Watches(&v1alpha1.Cluster{}, handler.EnqueueRequestsFromMapFunc(sameNameResource)).
+		Watches(&corev1.Secret{}, handler.EnqueueRequestsFromMapFunc(clusterSecretToCluster)).
```

Comment:
We could add a predicate here, right?
Check [bootstrap_controller](https://github.com/cloudoperators/greenhouse/blob/main/pkg/controllers/cluster/bootstrap_controller.go#L46-L48)

Reply:
As far I as check from the code, `Predicates` are used within `For(...)`. When you add one more than `For` block, manager fires and error:
```
For(...) should only be called once, could not assign multiple objects for reconciliation
```

Reply:
But we could use the Predicate in `Watches`, right?
See e.g. [helm_controller](https://github.com/cloudoperators/greenhouse/blob/main/pkg/controllers/plugin/helm_controller.go#L100-L101)

Reply:
Looking at the code we could think about moving this [availability check](https://github.com/cloudoperators/greenhouse/pull/526/files#diff-37497b2ef5a21a23aef197ccd85bd0f4ea2c65e7ed21023009e1f8191a986ca9R66-R69) into it's own predicate as [such](https://github.com/cloudoperators/greenhouse/blob/c7bb419fc0437f242fbfad58f09f2933da7d9be1/pkg/clientutil/predicates.go#L47-L54). This would ensure `ClusterKubeConfigs` are only enqueued if the cluster is ready.

This could reduce the reconciliations while keeping all relevant triggers from the conditions.